### ISSUE TITLE
Cannot link code using tbb::task_arena::automatic

### ISSUE THREAD

**Issue by rafbiels** [oneapi-src/oneTBB#529]
Since `tbb::task_arena::automatic` is defined as **non-inline** static member, it cannot be odr-used:
https://github.com/oneapi-src/oneTBB/blob/v2021.3.0/include/oneapi/tbb/task_arena.h#L215
meaning that the symbol cannot be used in a library linked against TBB, because it results in a linker error.

Minimal reproducer: https://godbolt.org/z/nT3rzf76a

This would work if  `tbb::task_arena::automatic` was either `inline` or initialised in the implementation file instead of the header.

It is worth noting that any compiler optimisation inlines the definition (try adding -O1 in the godbolt example) so the problem is only observed in debug (unoptimised) builds of the software stack I work on.

**Comment by alexey-katranov** [comment]
Consider workaround:
```
    m_arena = std::make_unique<tbb::task_arena>(int(tbb::task_arena::automatic), 0);
```
https://godbolt.org/z/robTGfYrM

**Comment by rafbiels** [comment]
Thank you, that's a good workaround - I'll use it for now.

**Comment by omalyshe** [comment]
@rafbiels Can you close the issue as resolved?

**Comment by mologie** [comment]
Could you reopen this @nofuturre? The bug still exists in the latest stable release and in master, above there was only a workaround posted.

Reference for affected line: https://github.com/uxlfoundation/oneTBB/blob/5ce1cc8/include/oneapi/tbb/task_arena.h#L253

**Comment by kboyarinov** [comment]
@mologie, thank you for drawing our attention to this issue. 

The fix #2120 resolves this issue for C++17 and later. However, for C++11 and C++14, the workaround described above will still be required. Would that be acceptable for your use case?

**Comment by mologie** [comment]
@kboyarinov yes, though technically the bug is not resolved though given earlier C++ versions are still supported by oneTBB, so not sure whether this issue should be closed (unless you aim to drop pre-C++17 support eventually).

I suppose a proper fix would have to define an out of class definition regardless of the C++ version TBB is compiled for (so that a C++17-compiled TBB works with C++14), and substitute it with a constexpr after the fact for C++17 or later with a `__TBB_CPP17_PRESENT` check.
